articles/integrate-tailwindcss-into-flask/index.md

```diff
@@ -0,0 +1,232 @@
+
+```
+
+The code block above creates a simple server with one endpoint (or route) that simple renders (or displays) a web page, which is supposed to be from a file called `index.html` as seen in the code. However, we haven't created that file yet. Remember, I mentioned that all HTML files go in the templates folder we created earlier. Hence, we'll go to that folder and add a new file called `index.html` so our flask server can see it. In `index.html` we can add a simple "hello world" h1 tag as follows:
+
+```hmtl
+<!DOCTYPE html>
+<html>
+<head>
+	<title></title>
+</head>
+<body>
+	<h1>Hello World</h1>
+</body>
+</html>
+```
+
+We'll save that and test run our server, from the terminal using the following command.
+
+```bash
+python app.py
+```
+Once the server is running head over to 
+
+`http://localhost:5000/index` to test the server and you should see the "hello world" from the `index.html` file.
+
+![demo1](/engineering-education/integrate-tailwindcss-into-flask/demo1.jpeg)
+
+## Installing TailwindCss
+Now that our server is up and running, let's install Tailwindcss. To install tailwind we can use either yarn or npm (depending on which one you have on your system). We  navigate to the `static` folder we created earlier from the terminal using the `cd` command, once there we'll install TailwindCss in there, as follows:
+
+```bash
+npm install tailwindcss
+```
+
+or with yarn:
+
+```bash
+yarn add tailwindcss
+```
+
+If you don't have npm you can install nodeJs since it comes with npm, or you can look at the yarn docs [here](https://classic.yarnpkg.com/en/docs/install#windows-stable) under the alternatives section. 
+Once this is done you should see that the installation includes some folders in the `static` folder, some of which include:
+
+* node_modules/ : which is where the dependencies for tailwind reside
+
+* package.json
+
+* yarn.lock (I used yarn)
+
+## Adding TailwindCss to our Flask App
+The first step is to add some of the required files in order to get tailwindcss to work. To make things easier, we'll create a new folder called `src` in the `static` folder from earlier, and in the `src` folder we'll add a file called `style.css`, and this is where tailwindCss directives will go, as we will see in a bit. To keep things clear the project folder structure should look as follows:
+
+```
+projectfolder/
+   - app.py
+   - templates/
+   - static/
+     - src/
+       - style.css
+```
+
+In this `style.css` file we'll add the following as stipulated by the tailwindcss [docs](https://tailwindcss.com/docs/installation) 
+
+```
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+These are called preprocessor directives by the documentation, and you can think of them as you would preprocessor directives from a language like C++, these directives essentially paste in a lot! of utility CSS classes during compile time (A lot!). The resulting CSS from this process is what we will link to our flask templates for styling.
```

Comment:
It would flow better as --> According to the documentation, these are called preprocessor directives. You can think of them as...